test/test_package.cpp

```diff
@@ -1339,41 +1339,61 @@ TEST(DDPackageTest, exactlyOneComparison) {
+        auto zeroState = dd->makeZeroState(nrQubits);
+
+        // Definition global operators
+        auto singleSiteX = dd->makeGateDD(dd::Xmat, 1, 0);
+        auto globalX     = singleSiteX;
+
+        auto singleSiteZ = dd->makeGateDD(dd::Zmat, 1, 0);
+        auto globalZ     = singleSiteZ;
+
+        auto singleSiteHadamard = dd->makeGateDD(dd::Hmat, 1, 0);
+        auto globalHadamard     = singleSiteHadamard;
+
+        for (int i = 1; i < nrQubits; i++) {
+            globalX        = dd->kronecker(globalX, singleSiteX);
+            globalZ        = dd->kronecker(globalZ, singleSiteZ);
+            globalHadamard = dd->kronecker(globalHadamard, singleSiteHadamard);
+        }
 
-TEST(DDPackageTest, stateFromVectorEmpty) {
-    auto dd = std::make_unique<dd::Package<>>(1);
-    auto v  = std::vector<std::complex<dd::fp>>{};
-    EXPECT_EQ(dd->makeStateFromVector(v), dd::vEdge::one);
+        // Global Expectation values
+        EXPECT_EQ(dd->expectationValue(globalX, zeroState), 0);
+        EXPECT_EQ(dd->expectationValue(globalZ, zeroState), 1);
+        EXPECT_EQ(dd->expectationValue(globalHadamard, zeroState), std::pow(dd::SQRT2_2, nrQubits));
+    }
 }
 
-TEST(DDPackageTest, stateFromVectorNoPowerOfTwo) {
-    auto dd = std::make_unique<dd::Package<>>(3);
-    auto v  = std::vector<std::complex<dd::fp>>{1, 2, 3, 4, 5};
-    EXPECT_THROW(dd->makeStateFromVector(v), std::invalid_argument);
+TEST(DDPackageTest, expectationValueLocalOperators) {
+    int max_qubits = 3;
+    for (dd::Qubit nrQubits = 1; nrQubits < max_qubits + 1; nrQubits++) {
+        auto dd        = std::make_unique<dd::Package<>>(nrQubits);
+        auto zeroState = dd->makeZeroState(nrQubits);
+
+        // Local expectation values at each site
+        for (int site = 0; site < nrQubits - 1; site++) {
+            // Definition local operators
+            auto xGate    = dd->makeGateDD(dd::Xmat, nrQubits, site);
+            auto zGate    = dd->makeGateDD(dd::Zmat, nrQubits, site);
+            auto hadamard = dd->makeGateDD(dd::Hmat, nrQubits, site);
+
+            EXPECT_EQ(dd->expectationValue(xGate, zeroState), 0);
+            EXPECT_EQ(dd->expectationValue(zGate, zeroState), 1);
+            EXPECT_EQ(dd->expectationValue(hadamard, zeroState), dd::SQRT2_2);
+        }
+    }
 }
 
-TEST(DDPackageTest, stateFromScalar) {
-    auto dd = std::make_unique<dd::Package<>>(1);
-    auto s  = dd->makeStateFromVector({1});
-    EXPECT_EQ(s.p->v, -1);
-    EXPECT_EQ(s.w.r->value, 1);
-    EXPECT_EQ(s.w.i->value, 0);
+TEST(DDPackageTest, expectationValueExceptions) {
+    const dd::Qubit nrQubits  = 1;
+    auto            dd        = std::make_unique<dd::Package<>>(nrQubits);
+    auto            zeroState = dd->makeZeroState(nrQubits);
+
+    dd         = std::make_unique<dd::Package<>>(nrQubits + 1);
+    auto xGate = dd->makeGateDD(dd::Xmat, nrQubits + 1, 0);
+    EXPECT_ANY_THROW(dd->expectationValue(xGate, zeroState));
 }
```

Comment:
The reason you are seeing a segmentation fault in the tests is that you create two packages here and when creating the second one, everything from the first one is gone.
This test should most likely look something like the following
```suggestion
TEST(DDPackageTest, expectationValueExceptions) {
    const dd::QubitCount nrQubits  = 2;
    
    auto dd = std::make_unique<dd::Package<>>(nrQubits);

    const auto zeroState = dd->makeZeroState(static_cast<dd::QubitCount>(nrQubits-1));
    const auto xGate = dd->makeGateDD(dd::Xmat, nrQubits, 0);
    
    EXPECT_ANY_THROW(dd->expectationValue(xGate, zeroState));
}
```
Some of the casting and type changes are to please the linter.

Reply:
Thanks! I was wondering if the package was the problem somehow